lib/si-data-nats/src/jetstream/context.rs

```diff
@@ -0,0 +1,109 @@
+use async_nats::jetstream::stream::{Config, DeleteStatus, Stream};
+use async_nats::{jetstream, HeaderMap};
+use bytes::Bytes;
+use telemetry::prelude::*;
+
+use crate::jetstream::{Consumer, JetstreamResult};
+use crate::Client;
+use crate::HeaderName;
+
+/// Used in a [`HeaderMap`] for consumers of stream to reply back to requesters (via a reply mailbox and outside of the
+/// stream).
+pub static REPLY_SUBJECT_HEADER_NAME: HeaderName = HeaderName::from_static("X-Reply-Subject");
+
+const DEFAULT_MAX_MESSAGES: i64 = 10_000;
+
+/// A wrapper around [`jetstream::Context`].
+#[derive(Debug)]
+pub struct Context {
+    inner: jetstream::Context,
+}
+
+impl Context {
+    /// Creates a new [`Context`].
+    pub fn new(client: Client) -> Self {
+        Self {
+            inner: jetstream::new(client.inner),
+        }
+    }
+
+    /// Finds or creates a stream.
+    pub async fn get_or_create_stream(
+        &self,
+        name: impl Into<String>,
+        subjects: Vec<String>,
+    ) -> JetstreamResult<Stream> {
+        let stream = self
+            .inner
+            .get_or_create_stream(Config {
+                name: name.into(),
+                subjects,
+                max_messages: DEFAULT_MAX_MESSAGES,
+                // TODO(nick): for the rebaser, we must have a work queue retention policy. This is temporary, just for
+                // the first pass. I hope this comment doesn't come back to haunt me.
+                // retention: RetentionPolicy::WorkQueue,
+                ..Default::default()
```

Comment:
Referencing the message earlier about work queues, this is the other half: when we get or create the stream, we need to decide it's a work queue. For this PR, we just use the defaults.